### PR TITLE
Bicep compiler merge

### PR DESCRIPTION
**Pull request by sk593**
# Description

This PR contains all the changes for the Bicep compiler merge. The branch has had smaller PRs opened for specific features that can be found below: 
  
- [x] Bicep types generator is updated to use the `bicep-types` repository as a submodule. This includes updating the types contract and adding support for resource methods (eg `listSecrets`): https://github.com/radius-project/radius/commit/43403fd165806e02cd941693a88356beaca186e5 . There was also a small update to how we handle flags on the `location` [property](https://github.com/radius-project/radius/pull/7662/commits/19afe967e2b19bb140c3879e7b923124d9cc2695)
- [x] Update to build workflow to initialize the submodule and publish types to `biceptypes.azurecr.io` using the `bicep publish-extension` command: https://github.com/radius-project/radius/commit/fed2462c1c2489a86880d576fca2a0deb4a28fb6   
- [x] Update validate bicep workflow to validate bicep files in the repository with the official Bicep compiler. This includes updating extension imports of all bicep files and adding a `bicepconfig.json` to the repository: https://github.com/radius-project/radius/pull/7702 
- [x] Update functional test workflow to publish types to `biceptypes.azurecr.io`. This includes publishing the bicep types to a PR tag and adding a temporary `bicepconfig.json` to consume the PR-specific types [https://github.com/radius-project/radius/pull/7730]
- [x] Update release workflow to publish types to a release-specific tag. This includes running a build in the `bicep-types-aws` repo to publish AWS types to a release tag 
- [x] Updates to `rad bicep download`. This command will download the official Bicep binary instead of the Radius one. The install script calls this command so the official Bicep binary will be downloaded on a Radius install: https://github.com/radius-project/radius/commit/bb8da88b8a54c49ba239a6b9eb75e8ba8d6952f1   
- [x] Updates to `rad init` to generate a default `bicepconfig.json`. This will pull the Radius and AWS types in our ACR with the current release tag: https://github.com/radius-project/radius/pull/7664

## Type of change

- This pull request fixes a bug in Radius and has an approved issue (issue link required).
- This pull request adds or changes features of Radius and has an approved issue (issue link required).
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: #issue_number

---------

# Description

_Please explain the changes you've made._

## Type of change

- This pull request fixes a bug in Radius and has an approved issue (issue link required).
- This pull request adds or changes features of Radius and has an approved issue (issue link required).
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: #issue_number

### CODE REVIEW

**Review by ytimocin — DISMISSED** [review]

File: .vscode/launch.json [hunk 1]
```diff
@@ -87,6 +87,21 @@
                 "RADIUSBACKENDURI": "http://localhost:9000",
             },
         },
+        {
+            "name": "Debug Jest Tests",
```
[the remark this answers is not in the document]
Comment (sk593) [reply .vscode/launch.json:91]: Yes, this is for the generator integration tests. I can rename it so its clearer

File: hack/bicep-types-radius/src/autorest.bicep/test/integration/generated/basic/test.rp1/2021-10-31/docs/testtype1.md
[the remark this answers is not in the document]
Comment (sk593) [reply hack/bicep-types-radius/src/autorest.bicep/test/integration/generated/basic/test.rp1/2021-10-31/docs/testtype1.md:1]: technically no, it's just the name of the test resource

File: bicepconfig.json
[the remark this answers is not in the document]
Comment (sk593) [reply bicepconfig.json:1]: Recompiled but this seems to be formatted correctly

File: .github/workflows/functional-test-noncloud.yaml
[the remark this answers is not in the document]
Comment (sk593) [reply .github/workflows/functional-test-noncloud.yaml:1]: Good idea, I was looking into that but the steps are not universal to each workflow. So it'd need a little more thought as to which parts go into the action vs are unique to each workflow. I'll add this as a possible cleanup task on the issue

File: .github/workflows/functional-test-noncloud.yaml [hunk 2]
```diff
@@ -126,7 +128,7 @@ jobs:
           echo "REL_VERSION=pr-${UNIQUE_ID}" >> $GITHUB_OUTPUT
           echo "DE_IMAGE=${{ env.DE_IMAGE }}" >> $GITHUB_OUTPUT
           echo "DE_TAG=${{ env.DE_TAG }}" >> $GITHUB_OUTPUT
-
+
```
[the remark this answers is not in the document]
Comment (sk593) [reply .github/workflows/functional-test-noncloud.yaml]: updated

File: .vscode/launch.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (sk593) [reply .vscode/launch.json]: Updated

File: .github/workflows/functional-test-noncloud.yaml
[the remark this answers is not in the document]
Comment (ytimocin) [reply .github/workflows/functional-test-noncloud.yaml:1]: Sounds good

**Review by ytimocin — DISMISSED** [review]
Needs a rebase

**Review by kachawla — DISMISSED** [review]

File: .github/workflows/functional-test-noncloud.yaml
[the remark this answers is not in the document]
Comment (ytimocin) [reply .github/workflows/functional-test-noncloud.yaml:1]: Have we created a follow-up issue for this work?

**Review by ytimocin — APPROVED** [review]

**Pull request merged**